1/ToDoApp/src/backend/resolvers/todo/todos.spec.js

```diff
@@ -0,0 +1,168 @@
+
+const server = new ApolloServer({typeDefs, resolvers});
+const { query, mutate } = createTestClient(server);
+
+describe('query', () => {
+    describe('todos', () => {
+        it('todo_list has 2 initial todos', async () => {
+            const res = await query({
+                query: GET_ALL_TODOS
+            });
+            expect(res.data.allTodos).toHaveLength(2);
+        });
+
+        it('returns todo-message for given id', async () => {
+            const res = await query({
+                query: GET_FIRST_TODOMESSAGE
+            });
+            expect(res.data.todoById).toMatchObject({message: "first todo"});
+        });
+    });
+});
+
+describe('mutate', () => {
+    var logInToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZW1haWwiOiJ5b3VyQGVtYWlsLmNvbSIsImlhdCI6MTU3NTQ0NjYzNCwiZXhwIjoxNjA3MDA0MjM0fQ.cHeP2Ih8Dxeyyxw3rePvfVCeJJFFnO7A9BrV1QL4hcA";
+    describe('given user is not logged in/no token', () => {
+        it('creating a todo returns an error', async () => {
+            const res = await mutate({
+                mutation: CREATE_TODO,
+                variables: {
+                    message: "neues Todo",
+                    token: ""
+                }
+            });
+            expect(res.errors).toHaveLength(1);
+        });
+        it('logIn with correct credentials returns token-String', async () => {
+            const res = await mutate({
+                mutation: LOGIN,
+                variables: {
+                    email: "your@email.com",
+                    password: "password"
+                }
+            });
+            expect(res.data.login).toBeDefined();
+        });
+    });
+    describe('give user is loggedIn', () => {
+        it('adds a todo', async () => {
+            const res_allTodos=await query ({
+                    query: GET_ALL_TODOS
+            });
+            const res = await mutate({
+                mutation: CREATE_TODO,
+                variables: {
+                    message: "neues Todo",
+                    token: logInToken
+                }
+            });
+            expect(res.data.addTodo).toHaveLength(res_allTodos.data.allTodos.length+1);
+            expect(res.data.addTodo[res.data.addTodo.length-1].message).toEqual("neues Todo");
+
+        });
+        describe('Modifying Todos', () => {
+            describe('given the loggedIn-User is the creator of the todo (allowed to modify)', () => {
+
+                it('updates todo message ', async () => {
+                    const res = await mutate({
+                        mutation: UPDATE_TODO_MESSAGE,
+                        variables: {
+                            id: 1,
+                            message: "kleines Update",
+                            finished: false,
+                            token: logInToken
+                        }
+                    });
+                    expect(res).toMatchObject({
```

Comment:
Feel free to use the `await expect(somePromise).resolves.toMatchObject(object)` syntax, it's nice! :slightly_smiling_face: 

https://jestjs.io/docs/en/tutorial-async#resolves